field/field.go

```diff
@@ -3,11 +3,17 @@ package field
 type Field interface {
 	Spec() *Spec
 	SetSpec(spec *Spec)
+
 	Pack() ([]byte, error)
 	Unpack(data []byte) (int, error)
 
-	SetBytes(b []byte)
-	Bytes() []byte
+	SetBytes(b []byte) error
+	Bytes() ([]byte, error)
+
+	String() (string, error)
+}
 
-	String() string
+type Struct interface {
```

Comment:
think it's worth finding a better name for this, and maybe adding a comment to explain its use. If I understand correctly, it basically just distinguishes between a field with subfields and a field without subfields, right? So maybe `StructuredField`? In that case might be good to indicate the relationship between the two:

```
type StructuredField interface {
	Field
	SetData(d interface{}) error
	Data() interface{}
}
```
Also, this inherits `field.Field`'s stuttering, so maybe just `Structured`, so that its `field.Structured`?

Reply:
I'm not happy with `Struct` either - I think `Structured` is an improvement on this although I'm also considering `DataReadWriter`. I'd like to limit interface embedding if possible just to restrict the scope of the interfaces we introduce into the public API. 

I'll message you offline to discuss further

Reply:
In the end, I just merged everything into a single `Field` interface so this convo is now redundant.